File: app/src/main/java/com/divercity/app/features/profile/profileprompt/ProfilePromptFragment.java
package com.divercity.app.features.profile.profileprompt;


import android.arch.lifecycle.ViewModel;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;

import com.divercity.app.R;
import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.databinding.FragmentProfilePromptBinding;

public class ProfilePromptFragment extends BaseFragment {

    ProfilePromptViewModel viewModel;
    FragmentProfilePromptBinding binding;

    public ProfilePromptFragment() {
        // Required empty public constructor
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_profile_prompt;
    }

    @Override
    public ViewModel getViewModel() {
        return null;
    }

    @Override
    public int getBindingVariable() {
        return -1;
    }

    public static ProfilePromptFragment newInstance() {
        return new ProfilePromptFragment();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding = (FragmentProfilePromptBinding) getViewDataBinding();

        binding.btnNext.setOnClickListener(view1 ->
                navigator.navigateToSelectUserType(getActivity(), true)
        );

        binding.btnNotNow.setOnClickListener(view1 ->
                navigator.navigateToSelectUserType(getActivity(), false)
        );
    }
}

File: app/src/main/java/com/divercity/app/features/onboarding/profileprompt/ProfilePromptFragment.kt
package com.divercity.app.features.onboarding.profileprompt

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.view.View
import com.divercity.app.R
import com.divercity.app.core.base.BaseFragment
import kotlinx.android.synthetic.main.fragment_profile_prompt.*

/**
 * Created by lucas on 26/10/2018.
 */

class ProfilePromptFragment : BaseFragment() {

    lateinit var viewModel: ProfilePromptViewModel

    companion object {

        fun newInstance(): ProfilePromptFragment = ProfilePromptFragment()
    }

    override fun layoutId(): Int = R.layout.fragment_profile_prompt

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProviders.of(this, viewModelFactory)[ProfilePromptViewModel::class.java]
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        btn_next.setOnClickListener {
            navigator.navigateToNextOnboarding(activity!!,
                    viewModel.accountType,
                    0,
                    false
            )
        }

        btn_not_now.setOnClickListener {
            navigator.navigateToHomeActivity(activity!!)
        }
    }

}